test/e2e/apimachinery/discovery.go

```diff
@@ -0,0 +1,55 @@
+package apimachinery
+
+import (
+	utilversion "k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+)
+
+var storageVersionServerVersion = utilversion.MustParseSemantic("v1.13.99")
+var _ = SIGDescribe("Discovery", func() {
+	var context *certContext
+	_ = context
+	f := framework.NewDefaultFramework("discovery")
+
+	var namespaceName string
+
+	BeforeEach(func() {
+		namespaceName = f.Namespace.Name
+
+		framework.SkipUnlessServerVersionGTE(storageVersionServerVersion, f.ClientSet.Discovery())
+
+		By("Setting up server cert")
+		context = setupServerCert(namespaceName, serviceName)
+	})
+
+	AfterEach(func() {
+		// TODO: cleanup?
+	})
+
+	It("[Feature: StorageVersionHash] Custom resource should have storage version hash", func() {
```

Comment:
I sent https://github.com/kubernetes/test-infra/pull/10881 to run this test in the alpha feature test suite.